pennylane/math/multi_dispatch.py

```diff
@@ -729,7 +729,7 @@ def scatter_element_add(tensor, index, value, like=None):
     return np.scatter_element_add(tensor, index, value, like=like)
 
 
-def unwrap(values, max_depth=None):
+def unwrap(values, max_depth=None, keep_type=False):

```

Comment:
`keep_type` makes it so that the resulting number is a numpy array with the same precision. Will help us distinguish between float32 and float64.